packages/material-ui/src/Switch/Switch.test.js

```diff
@@ -31,6 +31,15 @@ describe('<Switch />', () => {
     refInstanceof: window.HTMLSpanElement,
   }));
 
+  /* TODO Switch violates root component
```

Comment:
Oh yeah, I had forgotten about this one! :/ I have spent an afternoon trying to simplify the logic. I failed.